test/gjtests/test_natrans.py

```diff
@@ -0,0 +1,114 @@
+from neuron import h
+pc = h.ParallelContext()
+rank = pc.id()
+nhost = pc.nhost()
+pc.nthread(4)
+
+# Want to exercise the internal indexing schemes. So need mpi and threads.
+# Random order of sgid calls to source_var. Random order and location
+# of sgid source . Random order and location of sgid target. An sgid source
+# has random number of targets.
+# Might be a good idea to add some voltage sources and ki sources.
+
+ncell = 100
+nsrc = 50 # so that number of sgid pointing to _ref_nai
+          # multiple sgid may NOT point to same _ref_nai
+ntarget = 200 # so sgid on average sends to ntarget/nsrc targets.
+
+h('''
+begintemplate Cell
+public soma
+create soma
+proc init() {
+  soma.diam = 5
+  soma.L = 5
+  soma {
+    insert na_ion
+    ion_style("na_ion", 1,0,0,0,0) // finitialize leaves nai as is
+      // Otherwise cannot pass interpreter setting onto coreneuron
+      // unless use a mod file that WRITE nai in INITIAL via parameter
+  }
+}
+endtemplate Cell
+''')
+
+def test_natrans():
+  gids = [gid for gid in range(rank, ncell, nhost)]
+  cells = []
+  for gid in range(rank, ncell, nhost):
+    pc.set_gid2node(gid, rank)
+    cells.append(h.Cell())
+    pc.cell(gid, h.NetCon(cells[-1].soma(.5)._ref_v, None, sec=cells[-1].soma))
+
+  r = h.Random()
+  r.Random123(1,1,0)
+
+  #nsrc unique random sgids in range(ncell). Sample without replacement.
+  sgids = []
+  v = list(range(ncell))
+  for i in range(nsrc):
+    x = int(r.discunif(0, len(v)-1))
+    sgids.append(v[x])
+    v.pop(x)
+  del v
+  #print(sgids)
```

Comment:
just nitpick : remove print

Reply:
Removed this and the other prints. But one is activated in case of an error.